feat: Update Postgres search docs (#39183)
supabase/supabase: apps/docs/content/guides/database/full-text-search.mdx

  .select()
  .textSearch('description', 'green eggs', { type: 'websearch' })
```

</TabPanel>
</Tabs>

### Quoted phrases

Use quotes to search for exact phrases:

```sql
select * from books
where to_tsvector(description || ' ' || title) @@ websearch_to_tsquery('english', '"Green Eggs"');
-- Matches documents containing "Green" immediately followed by "Eggs"
```

### OR searches

Use "or" (case-insensitive) to search for multiple terms:

```sql
select * from books
where to_tsvector(description) @@ websearch_to_tsquery('english', 'puppy or rabbit');
-- Matches documents containing either "puppy" OR "rabbit"
```

### Negation

Use a dash (-) to exclude terms:

```sql
select * from books
where to_tsvector(description) @@ websearch_to_tsquery('english', 'animal -rabbit');
-- Matches documents containing "animal" but NOT "rabbit"
```

### Complex queries

Combine multiple operators for sophisticated searches:

```sql
select * from books
where to_tsvector(description || ' ' || title) @@
  websearch_to_tsquery('english', '"Harry Potter" or "Dr. Seuss" -vegetables');
-- Matches books by "Harry Potter" or "Dr. Seuss" but excludes those mentioning vegetables
```

## Creating indexes

> Now that we have Full Text Search working, let's create an `index`. This will allow Postgres to "build" the documents preemptively so that they

Reviewer: ```suggestion
Now that you have Full Text Search working, create an `index`. This allows Postgres to "build" the documents preemptively so that they
```

Nothing to do with you, but while we're here let's fix some things :)

</$Show>
</Tabs>

## Ranking search results [#ranking]

> Postgres provides ranking functions to sort search results by relevance, helping you present the most relevant matches first. Since ranking functions need to be computed server-side, we'll use RPC functions and generated columns.

Reviewer: ```suggestion
Postgres provides ranking functions to sort search results by relevance, helping you present the most relevant matches first. Since ranking functions need to be computed server-side, use RPC functions and generated columns.
```
